fiftyone/core/dataset.py

```diff
@@ -124,6 +124,8 @@ def load_dataset(name):
     Returns:
         a :class:`Dataset`
     """
+    if not (name and isinstance(name, str)):
```

Comment:
IMO this error would be more helpful/accurate:

```py
if not isinstance(name, str):
    raise ValueError("Dataset names must be strings")
```

`name==""` already produces an informative error